test/test_pipeline/components/data_preprocessing/test_categorical_imputation.py

```diff
@@ -34,15 +34,16 @@ def test_default_imputation(input_data_imputation, categorical):
         X = X.astype('str').astype('object')
         X[mask] = np.nan
     else:
-        imputation_value = 0
+        imputation_value = min(np.unique(X)) - 1
+
     Y = CategoricalImputation().fit_transform(X.copy())
-    assert ((np.argwhere(Y == imputation_value) == np.argwhere(mask)).all())
-    assert ((np.argwhere(Y != imputation_value) == np.argwhere(np.logical_not(mask))).all())
+
+    assert np.array_equal(Y == imputation_value, mask)
```

Comment:
How about np.testing.assert_array_equal ?

Reply:
Sure why not, probably gives better feedback on failure